### PR TITLE
Slack integration

### PR DESCRIPTION
**Pull request by Albertikuu**
Work in progress but it works!

### CODE REVIEW

File: app/controllers/commits_controller.rb [hunk 1]
```diff
@@ -15,6 +17,19 @@ def new
 
 	def create
 		@commit = Commit.create!(commit_params)
+		pretext= "#{current_user.full_name} has made a new commit in the category #{@commit.page.category.title.capitalize}"
+		title = "#{@commit.title}"
+		title_link = "https://basicshapes.herokuapp.com/bs/#{@commit.page.category.slug}/#{@commit.page.slug}"
+		footer = "with love by Basicshapes"
+        footer_icon = "https://avatars.slack-edge.com/2017-03-14/154308508964_76b45d8ff0cb5f68a8dd_72.jpg"
```
Comment (naoisegolden) [inline app/controllers/commits_controller.rb]: Seems indentation went wrong here!

File: app/controllers/commits_controller.rb [hunk 2]
```diff
@@ -15,6 +17,19 @@ def new
...
+        # add user icon: path & current_user.avatar_file_name
```
Comment (naoisegolden) [inline app/controllers/commits_controller.rb]: Are you planning on doing this @zlorine?

File: app/models/commit.rb [hunk 3]
```diff
@@ -6,4 +6,9 @@ class Commit < ApplicationRecord
     has_attached_file :image
   	validates_attachment_content_type :image, content_type: /\Aimage\/.*\z/
 
+
+def team
+	self.page.category.team
```
Comment (naoisegolden) [inline app/models/commit.rb]: indentation

File: .env.sample [hunk 4]
```diff
@@ -1,3 +0,0 @@
-# Copy this file to .env and put the right environment variables
-
-SLACK_API_TOKEN=
```
Comment (naoisegolden) [inline .env.sample]: 🤔 This file should not be deleted?

**Review by naoisegolden — CHANGES_REQUESTED** [review]
Hey! I left some comments.

File: app/controllers/commits_controller.rb (hunk 2 above)
Comment (zlorine) [inline app/controllers/commits_controller.rb]: yes, but we can't really do anything like that until we have set up the s3 thing, as I'd need the url path for the files

File: app/controllers/commits_controller.rb (hunk 1 above)
Comment (zlorine) [inline app/controllers/commits_controller.rb]: true that

File: .env.sample (hunk 4 above)
Comment (zlorine) [inline .env.sample]: do we need the .env.sample? i've made a new .env file

**Comment by zlorine** [comment]
Hi there, tried to answer your questions. Can't change indentation from this PC now but will do later from my laptop

File: .env.sample (hunk 4 above)
Comment (naoisegolden) [inline .env.sample]: the `.env` file is gitignored, the `.env.sample` is not gitignored, so when someone clones the repo can have a sample of the `.env` that they need, that's why it needs to stay in the repo.

File: app/controllers/commits_controller.rb (hunk 2 above)
Comment (naoisegolden) [inline app/controllers/commits_controller.rb]: Ok. Pro tip: create an issue in github and mention it in the code, e.g.:

    # TODO add user icon, see issue #5

**Pull request merged**